### PR TITLE
Empty state registered models

### PR DESCRIPTION
**Pull request by manaswinidas**
Closes: https://issues.redhat.com/browse/RHOAIENG-2229

## Description

Added empty state for Registered models

Screenshot 2024-04-08 at 5 22 32 PM

## How Has This Been Tested?

Change `empty` prop in `ModelRegistry.tsx`(there is no delete API yet) and navigate to `Model Registry` from sidebar

## Test Impact

Added e2e tests

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [X] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by manaswinidas** [comment]
@yih-wang Do we need the table header(filter and buttons) provided in the [mocks](https://www.figma.com/proto/KTsYhANYQqB0NUrXW7wQH1/Model-Registry?page-id=568%3A44455&type=design&node-id=5015-224143&viewport=2345%2C828%2C0.24&t=7VDnlaMsA8Cun4vu-1&scaling=min-zoom&starting-point-node-id=5015%3A224117)? There are several empty states do not have these. Here's an example(there's just an action button without the headers):
Screenshot 2024-04-05 at 10 21 13 PM

Also, now that the design is changing, do we still use the `PlusCircleIcon`, I tried using the Model Serving empty state image and it looks like this - let me know if we are still going ahead with the `PlusCircleIcon` for this:
Screenshot 2024-04-05 at 7 27 52 PM

**Comment by manaswinidas** [comment]
/hold 

Please merge #2677 before merging this PR

**Comment by yih-wang** [comment]
> Do we need the table header(filter and buttons) provided in the [mocks](https://www.figma.com/proto/KTsYhANYQqB0NUrXW7wQH1/Model-Registry?page-id=568%3A44455&type=design&node-id=5015-224143&viewport=2345%2C828%2C0.24&t=7VDnlaMsA8Cun4vu-1&scaling=min-zoom&starting-point-node-id=5015%3A224117)?

@manaswinidas  Yes, we need the table header in the case of Model Registry. The empty state only implies there are no models in the **current** registry. There could potentially be models in other registries, and we have to provide the ability for users to switch to them.

**Comment by yih-wang** [comment]
> Also, now that the design is changing, do we still use the PlusCircleIcon, I tried using the Model Serving empty state image and it looks like this - let me know if we are still going ahead with the PlusCircleIcon for this:

We will follow the old style (`PlusCircleIcon`) for now. The illustration that is used in the empty state varies between different components. The one you are using here is specifically for adding the model server. There is additional work from the design perspective to choose the proper illustration and adjust the color theme. So all the components (including Model Registry) that haven't been launched to the console are postponed to use the new style.

File: frontend/src/__mocks__/mockSingleModelRegistry.ts
Comment (dpanshug) [inline frontend/src/__mocks__/mockSingleModelRegistry.ts:19]: can you remove managedField, and make it as an empty array, it's not necessary for mocks

File: frontend/src/__mocks__/mockSingleModelRegistry.ts
Comment (dpanshug) [inline frontend/src/__mocks__/mockSingleModelRegistry.ts:1]:
```suggestion
export const mockModelRegistry = {
```
Change the file and function name

File: frontend/src/__mocks__/mockSingleModelRegistryRoute.ts
Comment (dpanshug) [inline frontend/src/__mocks__/mockSingleModelRegistryRoute.ts:17]: Same here, you can remove managedFields.

File: frontend/src/__mocks__/mockSingleModelRegistryRoute.ts
Comment (dpanshug) [inline frontend/src/__mocks__/mockSingleModelRegistryRoute.ts:1]:
```suggestion
export const mockModelRegistryRoute = {
```

File: frontend/src/__mocks__/mockSingleModelRegistryRoute.ts
Comment (dpanshug) [inline frontend/src/__mocks__/mockSingleModelRegistryRoute.ts:1]: You can move this to `mockRouteK8sResource.ts` similar to `mockRouteK8sResourceModelServing` function. check [here](https://github.com/dpanshug/odh-dashboard/blob/registered-table/frontend/src/__mocks__/mockRouteK8sResource.ts#L77)

File: frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts
Comment (dpanshug) [inline frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts:63]: These will be usefull for other test cases also, you might move these intercepts to `initIntercepts` and pass a variable for `size: 0` to mock for empty state.

File: frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts
Comment (dpanshug) [inline frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts:9]: You can use `visit` function below and change it's path from `/modelRegistry${modelRegistry}` to `/modelRegistry`

File: frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts
Comment (dpanshug) [inline frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts:65]: Use `visit` function here with path `/modelRegistry`

File: frontend/src/pages/modelRegistry/screens/EmptyRegisteredModels.tsx
Comment (dpanshug) [inline frontend/src/pages/modelRegistry/screens/EmptyRegisteredModels.tsx:18]:
```suggestion
      No models in selected registry
```
According to the [UX mock](https://www.figma.com/proto/KTsYhANYQqB0NUrXW7wQH1/Model-Registry?page-id=568%3A44455&type=design&node-id=5015-224143&viewport=2345%2C828%2C0.24&t=7VDnlaMsA8Cun4vu-1&scaling=min-zoom&starting-point-node-id=5015%3A224117&show-proto-sidebar=1)

File: frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx
Comment (dpanshug) [inline frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx:26]: Make this id specific to registered models, like `registered-model-page-title`

**Review by dpanshug — CHANGES_REQUESTED** [review]

File: frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx
Comment (manaswinidas) [reply frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx:26]: There were some failing tests due to which I added this. Removed it now.

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts
Comment (pnaik1) [inline frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts:10]: To tackle a cypress test flake , we are now using  `visit()` instead of `visitWithLogin()`
```suggestion
    cy.visit(`/modelRegistry`);
```

File: frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts
Comment (pnaik1) [inline frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts:35]: This function performs assertion
```suggestion
  shouldRegisteredModelsEmpty() {
    cy.findByTestId('no-registered-models').should('exist');
    return this;
  }
```

File: frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts
Comment (pnaik1) [inline frontend/src/__tests__/cypress/cypress/pages/modelRegistry.ts:37]: same here

**Review by pnaik1 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx
Comment (lucferbux) [inline frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx:20]: I'm gonna leave it for posterity, we should refactor this once we have the new redesign, I don't like this approach, we shouldn't render twice the component in the same view, In the near future we'll get the model registry in the header and the toolbar ONLY in the table

File: frontend/src/pages/modelRegistry/screens/EmptyRegisteredModels.tsx
Comment (lucferbux) [reply frontend/src/pages/modelRegistry/screens/EmptyRegisteredModels.tsx:15]: I still think you should use `EmptyStateHeader` here

**Review by lucferbux — CHANGES_REQUESTED** [review]
Just one nit, once that's fixed we can merge this!

**Review by lucferbux — CHANGES_REQUESTED** [review]
Just small nits on top of the later changes

**Comment by manaswinidas** [comment]
@lucferbux addressed the changes

File: frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts:36]: Please use `interceptOdh`, `interceptK8s` and `interceptK8sList` vs the plain vanilla `intercept` function.

**Review by dpanshug — CHANGES_REQUESTED** [review]

File: frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts
Comment (lucferbux) [reply frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts:15]: mmmm it shouldn't affect anything else right now, size is a default parameter, but you need to explicitly declare it in the mock data. The default value in the mock object is `4`, I would argue that maybe it's better put that value and explicitly declare something else if needed.
My only nit here is that size is not a descriptive name, i would call it `listSize` or something else.

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**
